tests/integration/test_client_server.py

```diff
@@ -109,3 +110,24 @@ def test_endpoint_not_found(mock_server):
             endpoint='fooo',
             hostport=mock_server.hostport,
         )
+
+
+@pytest.mark.gen_test
+def test_connection_close(mock_server):
+    tchannel = TChannel(name='test')
+
+    class TestHook(EventHook):
+        def before_send_request(self, request):
+            # close the connection
+            peer = tchannel._dep_tchannel.peers.get(mock_server.hostport)
+            peer.close()
+            peer.connections[0].closed = True
+
+    tchannel.hooks.register(TestHook())
```

Comment:
That's not the problem I'm trying to solve here. I'm trying to solve the hook being used to close the connection. Why not,

    mock_server.tchannel._dep_tchannel.close()

That will tell the server to immediately stop listening for new incoming connections, so when the `call` happens two lines below, it'll be trying to hit a hostport that doesn't exist.